validate.go

```diff
@@ -149,6 +153,52 @@ func formatKey(keys []string, parent reflect.Value) (string, error) {
 	return "", nil
 }
 
+func ConfigItemWhenValidation(v *validator.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	// validates that the when in config items references a valid item
+
+	root, ok := topStruct.Interface().(*RootConfig)
+	if !ok {
+		// this is an issue with the code and really should be a panic
+		return true
+	}
+
+	if fieldKind != reflect.String {
+		// this is an issue with the code and really should be a panic
+		return true
+	}
+
+	var whenValue string
+
+	whenValue = field.String()
+	if whenValue == "" {
+		return true
+	}
+
+	parts := strings.SplitN(whenValue, "=", 2)
```

Comment:
- weird to reuse `whenValue` variable for item name
- do we support `!=` cause this wont work for that?